app/clients/ethereum/ethereum.go

```diff
@@ -131,3 +131,35 @@ func (ec *Client) waitForTransactionReceipt(hash common.Hash) (txReceipt *types.
 
 	return ec.Client.TransactionReceipt(context.Background(), hash)
 }
+
+func (ec *Client) WaitBlocks(transactionHash string, blockNumber uint64) error {
```

Comment:
I think that it would be better if we return a channel that resolves once the transaction has more than `n` block confirmations. That way the "client" of this function will have to option to either to do blocking wait for this or continue with its logic.